MennaMohie/assignment2/Solution.cpp

```diff
@@ -0,0 +1,174 @@
+#include<bits/stdc++.h>
+using namespace std;
+
+template<class T>
+class BinaryTree;
+
+template<class T>
+class Node
+{
+    T value;
+    Node *left, *right;
+    friend class BinaryTree<T>;
+
+public:
+    Node():left(nullptr),right(nullptr) {}
+    Node(T val):left(nullptr),right(nullptr),value(val) {}
+
+    T getValue(){return value;}
+};
+
+template<class T>
+class BinaryTree
+{
+    Node<T> *root;
+
+public:
+    BinaryTree(T val):root(new Node<T>(val)) {}
+
+    Node<T>* getRoot()
+    {
+        return root;
+    }
+
+    bool contains(Node<T> *currentRoot, T val)
```

Comment:
Note that it may be more useful to return the node found if it exists, and nullptr otherwise.